In [1]:
import pandas as pd
import numpy as np
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

# Data Cleaning

Steps:
1. Determine Data Types
2. Deal with Missing Values (MAR/MNAR)
    - Continuous
    - Categorical
3. Outliers
    - Visual/Statistical
    - What to do with them? (Drop/winsorize/transform?)

In [2]:
df = pd.read_csv('nba_2012-2018_reg_data.csv')
df.head()

gameID  seasID      gmDate gmTime  seasTyp teamAbbr_A teamConf_A  \
0   10020    2012  2012-11-02  19:30  Regular        ATL       East   
1   10020    2012  2012-11-02  19:30  Regular        HOU       West   
2   10039    2012  2012-11-04  19:00  Regular        ATL       East   
3   10039    2012  2012-11-04  19:00  Regular        OKC       West   
4   10057    2012  2012-11-07  19:30  Regular        ATL       East   

   teamDiv_A teamLoc_A teamRslt_A  teamMin_A  teamDayOff_A  teamPTS_A  \
0  Southeast      Home       Loss        240             0        102   
1  Southwest      Away        Win        240             2        109   
2  Southeast      Away        Win        240             2        104   
3  Northwest      Home       Loss        240             2         95   
4  Southeast      Home        Win        240             3         89   

   teamAST_A  teamTO_A  teamSTL_A  teamBLK_A  teamPF_A  teamFGA_A  teamFGM_A  \
0         23        13         12          4        26         85         40   
1         22        21          8          2        18         90         38   
2         20        11         12          1        20         83         41   
3         27        21          4          9        21         71         33   
4         24        17          8          3        15         87         38   

   teamFG%_A  team2PA_A  team2PM_A  team2P%_A  team3PA_A  team3PM_A  \
0     0.4706         63         33     0.5238         22          7   
1     0.4222         60         30     0.5000         30          8   
2     0.4940         58         33     0.5690         25          8   
3     0.4648         49         24     0.4898         22          9   
4     0.4368         65         31     0.4769         22          7   

   team3P%_A  teamFTA_A  teamFTM_A  teamFT%_A  teamORB_A  teamDRB_A  \
0     0.3182         17         15     0.8824          7         29   
1     0.2667         29         25     0.8621         23         35   
2     0.3200         20         14     0.7000         12         26   
3     0.4091         22         20     0.9091          7         30   
4     0.3182         12          6     0.5000         17         34   

   teamTRB_A  teamPTS1_A  teamPTS2_A  teamPTS3_A  teamPTS4_A  teamPTS5_A  \
0         36          21          23          30          28           0   
1         58          28          25          28          28           0   
2         38          30          17          28          29           0   
3         37          22          29          23          21           0   
4         51          22          29          14          24           0   

   teamPTS6_A  teamPTS7_A  teamPTS8_A  teamTREB%_A  teamASST%_A  teamTS%_A  \
0           0           0           0      38.2979      57.5000     0.5515   
1           0           0           0      61.7021      57.8947     0.5304   
2           0           0           0      50.6667      48.7805     0.5664   
3           0           0           0      49.3333      81.8182     0.5887   
4           0           0           0      55.4348      63.1579     0.4822   

   teamEFG%_A  teamOREB%_A  teamDREB%_A  teamTO%_A  teamSTL%_A  teamBLK%_A  \
0      0.5118      16.6667      55.7692    12.3246     12.3226      4.1075   
1      0.4667      44.2308      83.3333    16.9683      8.2151      2.0538   
2      0.5422      28.5714      78.7879    10.7004     13.2351      1.1029   
3      0.5282      21.2121      71.4286    20.6530      4.4117      9.9263   
4      0.4770      36.1702      75.5556    15.5564      8.8711      3.3267   

   teamBLKR_A  teamPPS_A  teamFIC_A  teamFIC40_A  teamOrtg_A  teamDrtg_A  \
0      6.3492     1.2000     73.625      61.3542    104.7423    111.9305   
1      3.3333     1.2111     81.875      68.2292    111.9305    104.7423   
2      1.7241     1.2530     77.750      64.7917    114.7038    104.7775   
3     18.3673     1.3380     71.500      59.5833    104.7775    114.7038   
4      4.6154     1.0230   

In [3]:
df.tail()

gameID  seasID      gmDate gmTime  seasTyp teamAbbr_A teamConf_A  \
14753   16047    2016  2017-03-31  09:00  Regular        WAS       East   
14754   16496    2017  2017-12-04  09:00  Regular        UTA       West   
14755   16496    2017  2017-12-04  09:00  Regular        WAS       East   
14756   16751    2017  2018-01-10  07:00  Regular        UTA       West   
14757   16751    2017  2018-01-10  07:00  Regular        WAS       East   

       teamDiv_A teamLoc_A teamRslt_A  teamMin_A  teamDayOff_A  teamPTS_A  \
14753  Southeast      Away       Loss        241             1         88   
14754  Northwest      Home        Win        241             3        116   
14755  Southeast      Away       Loss        240             3         69   
14756  Northwest      Away        Win        240             3        107   
14757  Southeast      Home       Loss        241             3        104   

       teamAST_A  teamTO_A  teamSTL_A  teamBLK_A  teamPF_A  teamFGA_A  \
14753         12        15          8          1        23         87   
14754         29        16          9          3        20         76   
14755          9        13          8          2        21         80   
14756         23        13         15          3        17         92   
14757         25        23          8          1        17         78   

       teamFGM_A  teamFG%_A  team2PA_A  team2PM_A  team2P%_A  team3PA_A  \
14753         35     0.4023         70         30     0.4286         17   
14754         43     0.5658         42         27     0.6429         34   
14755         23     0.2875         60         19     0.3167         20   
14756         43     0.4674         59         32     0.5424         33   
14757         41     0.5256         53         31     0.5849         25   

       team3PM_A  team3P%_A  teamFTA_A  teamFTM_A  teamFT%_A  teamORB_A  \
14753          5     0.2941         16         13     0.8125         14   
14754         16     0.4706         18         14     0.7778         10   
14755          4     0.2000         23         19     0.8261         12   
14756         11     0.3333         13         10     0.7692         11   
14757         10     0.4000         17         12     0.7059          9   

       teamDRB_A  teamTRB_A  teamPTS1_A  teamPTS2_A  teamPTS3_A  teamPTS4_A  \
14753         31         45          19          20          22          27   
14754         42         52          29          35          24          28   
14755         19         31          15          15          22          17   
14756         28         39          25          21          37          24   
14757         36         45          32          18          27          27   

       teamPTS5_A  teamPTS6_A  teamPTS7_A  teamPTS8_A  teamTREB%_A  \
14753           0           0           0           0      51.7241   
14754           0           0           0           0      62.6506   
14755           0           0           0           0      37.3494   
14756           0           0           0           0      46.4286   
14757           0           0           0           0      53.5714   

       teamASST%_A  teamTS%_A  teamEFG%_A  teamOREB%_A  teamDREB%_A  \
14753      34.2857     0.4679      0.4310      28.5714      81.5789   
14754      67.4419     0.6911      0.6711      34.4828      77.7778   
14755      39.1304     0.3828      0.3125      22.2222      65.5172   
14756      53.4884     0.5475      0.5272      23.4043      75.6757   
14757      60.9756     0.6083      0.5897      24.3243      76.5957   

       teamTO%_A  teamSTL%_A  teamBLK%_A  teamBLKR_A  teamPPS_A  teamFIC_A  \
14753    13.7564      8.8292      1.1036      1.4286     1.0115     48.500   
14754    16.0128     10.2463      3.4154      7.1429     1.5263    108.750   
14755    12.6067      9.1078      2.2770      3.3333     0.8625     22.125   
14756    11.7413     15.2981      3.0596      5.0847     1.1630     84.625   
14757    21.2021      8.1590      1.019

In [4]:
df.shape

(14758, 211)

First, let's look at what each column means.

## Dataset/Column Definitions:

**Dataset Description**: This dataset contains data pertaining to game level information of regular season games in the NBA from the 2012-2013 season to the 2017-2018 season. It is formatted in such a way that each row is a game with information about the two teams that play. Each game has two entries, one from each team's perspective of the game.

**Column Definitions**:
- gameID: unique identification number for each regular season game from 2012-2013 season to 2017-2018 season (there are exactly two rows for each game from each team's perspective, ultimately we only want one instance of each game, but some alterations to the data must be made first) / also serves as part of primary key for dataset (in conjunction with 'teamAbbr_A')
- seasID: the starting year for the season (e.g. 2013 if the season takes place from 2013-2014) / acts as the differentiator between seasons, since using the year date of a game is not an appropriate separation here
- gmDate: the actual date the game is being played (Year/Month/Day)
- gmTime: the time the game begins on a 24 hour clock (ET)
- seasTyp: this indicates what part of the season it is, in this case they are all 'Regular' and therefore this column should be dropped
- teamAbbr_A: the name of team A (there are 30 of these, indicating 30 teams in the league) / serves as the second part of the primary key (in conjunction with the 'gameID')
- teamConf_A: the conference in which team A plays (either East or West)
- teamDiv_A: the division in which team A plays (there are 3 divisions in each conference, meaning there are 6 total divisions)
- teamLoc_A: the indicator for whether team A is playing 'Home' (in their city) or 'Away' (in team B's city)
- teamRslt_A: the outcome of the game from the perspective of team A (Win or Loss)
- teamMin_A: the accumulated minutes played in the game by the players (mostly 240 as that is how many minutes are played cumulatively by players for a regular 4 quarter game, i.e. 5 players play 4 quarters that are 12 minutes each results in 240 cumulative minutes)
- teamDayOff_A: the number of days without a game since the last game for team A (e.g. if the value is 1, team A played 2 days prior)
- teamPTS_A: the total points team A had in the game
- teamAST_A: the total assists team A had in the game
- teamTO_A: the total turnovers team A had in the game
- teamSTL_A: the total steals team A had in the game
- teamBLK_A: the total blocks team A had in the game
- teamPF_A: the total personal fouls team A had in the game
- teamFGA_A: the total attempted field goals (shots) team A had in the game
- teamFGM_A: the total made field goals (made shots) team A had in the game
- teamFG\%_A: the teamFGM / teamFGA for team A in the game 
- team2PA_A: the 2 point field goal attempts team A had in the game
- team2PM_A: the 2 point field goal makes team A had in the game
- team2P\%_A: the team2PM / team2PA for team A in the game
- team3PA_A: the 3 point field goal attempts team A had in the game
- team3PM_A: the 3 point field goal makes team A had in the game
- team3P\%_A: the team3PM / team3PA for team A in the game
- teamFTA_A: the number of free throw attempts by team A in the game
- teamFTM_A: the number of free throw makes by team A in the game
- teamFT\%_A: the teamFTM / teamFTA for team A in the game
- teamORB_A: the number of offensive rebounds team A had in the game
- teamDRB_A: the number of defensive rebounds team A had in the game
- teamTRB_A: the total number of rebounds team A had in the game
- teamPTS1_A: the total points team A had in the first quarter of the game
- teamPTS2_A: the total points team A had in the second quarter of the game
- teamPTS3_A: the total points team A had in the third quarter of the game
- teamPTS4_A: the total points team A had in the fourth quarter of the game
- teamPTS5_A: the total points team A had in the first overtime if the teams are tied after 4 quarters
- teamPTS6_A: the total points team A had in the second overtime if game is still tied after first overtime
- teamPTS7_A: the total points team A had in the third overtime if game is still tied after second overtime
- teamPTS8_A: the total points team A had in the fourth overtime if game is still tied after third overtime
- teamTREB\%_A: the percent of total rebounds team A had in the game (teamTRB_A / (teamTRB_A + teamTRB_B))
- teamASST\%_A: the percent of assisted field goal makes over all field goal makes for team A (teamAST_A / teamFGM_A)
- teamTS\%_A: the true shooting percentage statistic for team A (teamPTS_A / (2 * (teamFGA_A + (teamFTA_A * 0.44))))
- teamEFG\%_A: the effective field goal percentage statistic for team A (teamFGM_A + (0.5 * team3PM_A) / teamFGA_A)
- teamOREB\%_A: the percent of offensive rebounds team A had in the game (teamORB_A / (teamORB_A + teamORB_B))
- teamDREB\%_A: the percent of defensive rebounds team A had in the game (teamDRB_A / (teamDRB_A + teamDRB_B))
- teamTO\%_A: the turnover percentage for team A (teamTO_A * 100 / (teamFGA_A + 0.44 * teamFTA_A + teamTO_A))
- teamSTL\%_A: the steal percentage for team A (teamSTL_A * 100 / poss_A)
- teamBLK\%_A: the block percentage for team A (teamBLK_A * 100 / poss_A)
- teamBLKR_A: the block rate for team A (teamBLK_A * 100 / team2PA_B)
- teamPPS_A: the points per shot for team A (teamPTS_A / teamFGA_A)
- teamFIC_A: the floor impact counter statistic for team A (teamPTS_A + teamORB_A + 0.75 * teamDRB_A + teamAST_A + teamSTL_A + teamBLK_A – 0.75 * teamFGA_A – 0.375 * teamFTA_A – teamTO_A – 0.5 * teamPF_A)
- teamFIC40_A: the floor impact counter per 40 minutes by team A (((teamPTS_A + teamORB_A + 0.75 * teamDRB_A + teamAST_A + teamSTL_A + teamBLK_A – 0.75 * teamFGA_A – 0.375 * teamFTA_A – teamTO_A – 0.5 * teamPF_A) * 40 * 5) / teamMin_A)
- teamOrtg_A: the offensive rating for team A in the game (teamPTS_A * 100 / poss_A)
- teamDrtg_A: the defensive rating for team A in the game (teamPTS_B * 100 / poss_A)
- teamEDiff_A: the efficiency difference for team A in the game (teamOrtg_A – teamDrtg_A)
- teamPlay\%_A: the play percentage for team A in the game (teamFGM_A / (teamFGA_A – teamORB_A + teamTO_A))
- teamAR_A: the assist rate for team A in the game ((teamAST_A * 100) / (teamFGA_A – 0.44 * teamFTA_A + teamAST_A + teamTO_A)
- teamAST/TO_A: the assist to turnover ratio for team A in the game (teamAST_A / teamTO_A)
- teamSTL/TO_A: the steal to turnover ratio for team A in the game (teamSTL_A / teamTO_A)
- poss_A: the total possessions for team A (teamFGA_A – (teamORB_A / teamORB_A + teamDRB_B) * (teamFGA_A – teamFGM_A) * 1.07 + teamTO_A) + (0.4 * teamFTA_A)
- pace_A: the pace for team A in the game ((poss_A * 48 * 5) / teamMin_A))
- offLNm1: first official/referee last name
- offFNm1: first official/referee first name
- offLNm2: second official/referee last name
- offFNm2: second official/referee first name
- offLNm3: third official/referee last name
- offFNm3: third official/referee first name
- teamAbbr_B: the name of team B (there are 30 of these, indicating 30 teams in the league) / serves as another possible second part of the primary key (in conjunction with the 'gameID')
- teamConf_B: the conference in which team B plays (either East or West)
- teamDiv_B: the division in which team B plays (there are 3 divisions in each conference, meaning there are 6 total divisions)
- teamLoc_B: the indicator for whether team B is playing 'Home' (in their city) or 'Away' (in team A's city)
- teamRslt_B: the outcome of the game from the perspective of team B (Win or Loss)
- teamMin_B: the accumulated minutes played in the game by the players (mostly 240 as that is how many minutes are played cumulatively by players for a regular 4 quarter game, i.e. 5 players play 4 quarters that are 12 minutes each results in 240 cumulative minutes)
- teamDayOff_B: the number of days without a game since the last game for team B (e.g. if the value is 1, team B played 2 days prior)
- teamPTS_B: the total points team B had in the game
- teamAST_B: the total assists team B had in the game
- teamTO_B: the total turnovers team B had in the game
- teamSTL_B: the total steals team B had in the game
- teamBLK_B: the total blocks team B had in the game
- teamPF_B: the total personal fouls team B had in the game
- teamFGA_B: the total attempted field goals (shots) team B had in the game
- teamFGM_B: the total made field goals (made shots) team B had in the game
- teamFG\%_B: the teamFGM / teamFGA for team B in the game 
- team2PA_B: the 2 point field goal attempts team B had in the game
- team2PM_B: the 2 point field goal makes team B had in the game
- team2P\%_B: the team2PM / team2PA for team B in the game
- team3PA_B: the 3 point field goal attempts team B had in the game
- team3PM_B: the 3 point field goal makes team B had in the game
- team3P\%_B: the team3PM / team3PA for team B in the game
- teamFTA_B: the number of free throw attempts by team B in the game
- teamFTM_B: the number of free throw makes by team B in the game
- teamFT\%_B: the teamFTM / teamFTA for team B in the game
- teamORB_B: the number of offensive rebounds team B had in the game
- teamDRB_B: the number of defensive rebounds team B had in the game
- teamTRB_B: the total number of rebounds team B had in the game
- teamPTS1_B: the total points team B had in the first quarter of the game
- teamPTS2_B: the total points team B had in the second quarter of the game
- teamPTS3_B: the total points team B had in the third quarter of the game
- teamPTS4_B: the total points team B had in the fourth quarter of the game
- teamPTS5_B: the total points team B had in the first overtime if the teams are tied after 4 quarters
- teamPTS6_B: the total points team B had in the second overtime if game is still tied after first overtime
- teamPTS7_B: the total points team B had in the third overtime if game is still tied after second overtime
- teamPTS8_B: the total points team B had in the fourth overtime if game is still tied after third overtime
- teamTREB\%_B: the percent of total rebounds team B had in the game (teamTRB_B / (teamTRB_B + teamTRB_A))
- teamASST\%_B: the percent of assisted field goal makes over all field goal makes for team B (teamAST_B / teamFGM_B)
- teamTS\%_B: the true shooting percentage statistic for team B (teamPTS_B / (2 * (teamFGA_B + (teamFTA_B * 0.44))))
- teamEFG\%_B: the effective field goal percentage statistic for team B (teamFGM_B + (0.5 * team3PM_B) / teamFGA_B)
- teamOREB\%_B: the percent of offensive rebounds team B had in the game (teamORB_B / (teamORB_B + teamORB_A))
- teamDREB\%_B: the percent of defensive rebounds team B had in the game (teamDRB_B / (teamDRB_B + teamDRB_A))
- teamTO\%_B: the turnover percentage for team B (teamTO_B * 100 / (teamFGA_B + 0.44 * teamFTA_B + teamTO_B))
- teamSTL\%_B: the steal percentage for team B (teamSTL_B * 100 / poss_B)
- teamBLK\%_B: the block percentage for team B (teamBLK_B * 100 / poss_B)
- teamBLKR_B: the block rate for team B (teamBLK_B * 100 / team2PA_A)
- teamPPS_B: the points per shot for team B (teamPTS_B / teamFGA_B)
- teamFIC_B: the floor impact counter statistic for team B (teamPTS_B + teamORB_B + 0.75 * teamDRB_B + teamAST_B + teamSTL_B + teamBLK_B – 0.75 * teamFGA_B – 0.375 * teamFTA_B – teamTO_B – 0.5 * teamPF_B)
- teamFIC40_B: the floor impact counter per 40 minutes by team B (((teamPTS_B + teamORB_B + 0.75 * teamDRB_B + teamAST_B + teamSTL_B + teamBLK_B – 0.75 * teamFGA_B – 0.375 * teamFTA_B – teamTO_B – 0.5 * teamPF_B) * 40 * 5) / teamMin_B)
- teamOrtg_B: the offensive rating for team B in the game (teamPTS_B * 100 / poss_B)
- teamDrtg_B: the defensive rating for team B in the game (teamPTS_A * 100 / poss_B)
- teamEDiff_B: the efficiency difference for team B in the game (teamOrtg_B – teamDrtg_B)
- teamPlay\%_B: the play percentage for team B in the game (teamFGM_B / (teamFGA_B – teamORB_B + teamTO_B))
- teamAR_B: the assist rate for team B in the game ((teamAST_B * 100) / (teamFGA_B – 0.44 * teamFTA_B + teamAST_B + teamTO_B)
- teamAST/TO_B: the assist to turnover ratio for team B in the game (teamAST_B / teamTO_B)
- teamSTL/TO_B: the steal to turnover ratio for team B in the game (teamSTL_B / teamTO_B)
- poss_B: the total possessions for team B (teamFGA_B – (teamORB_B / teamORB_B + teamDRB_A) * (teamFGA_B – teamFGM_B) * 1.07 + teamTO_B) + (0.4 * teamFTA_B)
- pace_B: the pace for team B in the game ((poss_B * 48 * 5) / teamMin_B)))
- rank_A: comparative rank of team A in conference up to game date
- rankOrd_A: comparative rank of team A in conference up to game date
- gameWon_A: number of games team A won in the season up to game date
- gameLost_A: number of games team A lost in the season up to game date
- stk_A: current streak for team A in string format (e.g. W2 indicates team A won the last 2 games they've played)
- stkType_A: type of current streak for team A (win, loss, or -)
- stkTot_A: number of games in current streak for team A
- gameBack_A: number of games team A is behind first place in conference
- ptsFor_A: total accumulated points by team A up to the date of the game in the season
- ptsAgnst_A: total accumulated points scored against team A up to the date of the game in the season
- homeWin_A: number of home games won up to the point of the game for team A in a season
- homeLoss_A: number of home games lost up to the point of the game for team A in a season
- awayWin_A: number of away games won up to the point of the game for team A in a season
- awayLoss_A: number of away games lost up to the point of the game for team A in a season
- confWin_A: number of games won against teams in same conference up to the point of the game for team A in a season
- confLoss_A: number of games lost against teams in same conference up to the point of the game for team A in a season
- lastFive_A: number of wins in the last five games for team A prior to game
- lastTen_A: number of wins in the last ten games for team A prior to game
- gamePlay_A: number of games played up to the point of the game by team A
- ptsScore_A: average points scored by team A per game of all games in the season up to the point of this game
- ptsAllow_A: average points scored against team A per game of all games in the season up to the point of this game
- ptsDiff_A: points differential for the season up to the point of this game for team A (ptsScore_A - ptsAllow_A)
- opptGmPlay_A: accumulated games played by opponents of team A up to the point of this game in the season
- opptGmWon_A: accumulated games won by opponents of team A up to the point of this game in the season
- opptOpptGmPlay_A: accumulated games played by opponents of opponents of team A up to the point of this game in the season
- opptOpptGmWon_A: accumulated games won by opponents of opponents of team A up to the point of this game in the season
- sos_A: strength of schedule for team A up to the point of this game in the season (((2 * opptGmWon_A / opptGmPlay_A) + (opptOpptGmWon_A / opptOpptGmPlay_A)) / 3
- rel\%Indx_A: relative percent index for team A, measures a team's strength of schedule and how a team does against that schedule, up to the point of this game in the season ((0.25 * gamePlay_A / gameWon_A) + (0.5 * opptGmWon_A / opptGmPlay_A) + (0.25 * opptOpptGmWon_A / opptOpptGmPlay_A))
- mov_A: margin of victory for team A up to the point of this game in the season ((ptsFor_A – ptsAgnst_A) / gamePlay_A)
- srs_A: simple rating system statistic for team A up to the point of this game in the season (mov_A – sos_A)
- pw\%_A: projected winning percentage for team A for the season based on mov_A (((mov_A * 2.7) + 41) / 82)
- pyth\%13.91_A: Pythagorean 13.91 winning percentage for team A for the season based on games up to this game(ptsFor_A ^ 13.91 / (ptsFor_A ^ 13.91 + ptsAgnst_A ^ 13.91))
- wpyth13.91_A: Pythagorean 13.91 expected wins for team A for the season (pyth\%13.91_A * 82)
- lpyth13.91_A: Pythagorean 13.91 expected losses for team A for the season (82 – wpyth13.91_A)
- pyth%16.5_A: Pythagorean 16.5 winning percentage for team A for the season based on games up to this game(ptsFor_A ^ 16.5 / (ptsFor_A ^ 16.5 + ptsAgnst_A ^ 16.5))
- wpyth16.5_A: Pythagorean 16.5 expected wins for team A for the season (pyth\%16.5_A * 82)
- lpyth16.5_A: Pythagorean 16.5 expected losses for team A for the season (82 – wpyth16.5_A)
- rank_B: comparative rank of team B in conference up to game date
- rankOrd_B: comparative rank of team B in conference up to game date
- gameWon_B: number of games team B won in the season up to game date
- gameLost_B: number of games team B lost in the season up to game date
- stk_B: current streak for team B in string format (e.g. W2 indicates team B won the last 2 games they've played)
- stkType_B: type of current streak for team B (win, loss, or -)
- stkTot_B: number of games in current streak for team B
- gameBack_B: number of games team B is behind first place in conference
- ptsFor_B: total accumulated points by team B up to the date of the game in the season
- ptsAgnst_B: total accumulated points scored against team B up to the date of the game in the season
- homeWin_B: number of home games won up to the point of the game for team B in a season
- homeLoss_B: number of home games lost up to the point of the game for team B in a season
- awayWin_B: number of away games won up to the point of the game for team B in a season
- awayLoss_B: number of away games lost up to the point of the game for team B in a season
- confWin_B: number of games won against teams in same conference up to the point of the game for team B in a season
- confLoss_B: number of games lost against teams in same conference up to the point of the game for team B in a season
- lastFive_B: number of wins in the last five games for team B prior to game
- lastTen_B: number of wins in the last ten games for team B prior to game
- gamePlay_B: number of games played up to the point of the game by team B
- ptsScore_B: average points scored by team B per game of all games in the season up to the point of this game
- ptsAllow_B: average points scored against team B per game of all games in the season up to the point of this game
- ptsDiff_B: points differential for the season up to the point of this game for team B (ptsScore_B - ptsAllow_B)
- opptGmPlay_B: accumulated games played by opponents of team B up to the point of this game in the season
- opptGmWon_B: accumulated games won by opponents of team B up to the point of this game in the season
- opptOpptGmPlay_B: accumulated games played by opponents of opponents of team B up to the point of this game in the season
- opptOpptGmWon_B: accumulated games won by opponents of opponents of team B up to the point of this game in the season
- sos_B: strength of schedule for team B up to the point of this game in the season (((2 * opptGmWon_B / opptGmPlay_B) + (opptOpptGmWon_B / opptOpptGmPlay_B)) / 3
- rel\%Indx_B: relative percent index for team B, measures a team's strength of schedule and how a team does against that schedule, up to the point of this game in the season ((0.25 * gamePlay_B / gameWon_B) + (0.5 * opptGmWon_B / opptGmPlay_B) + (0.25 * opptOpptGmWon_B / opptOpptGmPlay_B))
- mov_B: margin of victory for team B up to the point of this game in the season ((ptsFor_B – ptsAgnst_B) / gamePlay_B)
- srs_B: simple rating system statistic for team B up to the point of this game in the season (mov_B – sos_B)
- pw\%_B: projected winning percentage for team B for the season based on mov_B (((mov_B * 2.7) + 41) / 82)
- pyth\%13.91_B: Pythagorean 13.91 winning percentage for team B for the season based on games up to this game(ptsFor_B ^ 13.91 / (ptsFor_B ^ 13.91 + ptsAgnst_B ^ 13.91))
- wpyth13.91_B: Pythagorean 13.91 expected wins for team B for the season (pyth\%13.91_B * 82)
- lpyth13.91_B: Pythagorean 13.91 expected losses for team B for the season (82 – wpyth13.91_B)
- pyth%16.5_B: Pythagorean 16.5 winning percentage for team B for the season based on games up to this game(ptsFor_B ^ 16.5 / (ptsFor_B ^ 16.5 + ptsAgnst_B ^ 16.5))
- wpyth16.5_B: Pythagorean 16.5 expected wins for team B for the season (pyth\%16.5_B * 82)
- lpyth16.5_B: Pythagorean 16.5 expected losses for team B for the season (82 – wpyth16.5_B)
- line_price_A: average betting line price for team A
- spread_A: average betting spread for team A
- spread_price_A: average betting spread price for team A
- total_A: average betting total points for team A
- total_price_A: average betting total points price for team A
- line_price_B: average betting line price for team B
- spread_B: average betting spread for team B
- spread_price_B: average betting spread price for team B
- total_B: average betting total points for team B
- total_price_B: average betting total points price for team B